File: Domain/Class.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Class
    {
        public Guid Id { get; set; }
        public String ClassName { get; set; }
        public Professor ClassProfessor { get; set; }
        public ICollection<Student> Students { get; set; }
        public ICollection<ClassProfessor> Professors { get; set; }
        public ICollection<ClassSubject> Subjects { get; set; }
    }
}

File: client-app/src/app/models/class.ts
import { Professor } from "./professor";
import { Profile } from "./profile";
import { Student } from "./student";
import { Subject } from "./subject";

export interface Class {
    id: string;
    className: string;
    classProfessor: Professor | null;
    subjects: Subject[];
    professors: Profile[];
    students: Student[];
}

export class Class implements Class {
    constructor(init?: ClassFormValues) {
        Object.assign(this, init);
    }
}

export class ClassFormValues {
    id?: string = undefined;
    className: string = '';

    constructor(classR?: ClassFormValues) {
        if (classR) {
            this.id = classR.id;
            this.className = classR.className;
        };
    }
}